web/src/engine/main/src/contextManagerBase.ts

```diff
@@ -274,8 +274,6 @@ export abstract class ContextManagerBase<MainConfig extends EngineConfiguration>
     // (!wasNull || !!keyboard) - blocks events for `null` -> `null` transitions.
     // (keyman/keymanweb.com#96)
     if(this.currentKeyboardSrcTarget() == originalKeyboardTarget && (!wasNull || !!keyboard)) {
-      // Perform standard context-reset ops, including the processing of new-context events.
-      this.resetContext();
```

Comment:
This call is during Promise resolution that's synchronous with another, later call.  (It's visible within this PR's changes - see edits in `keymanEngine.ts` + CTRL+F `doContextReset`.)  That later call occurs during OSK layout-batching mode and _also_ triggers new-context rule processing, and so the later one is the one we wish to keep.